Speckle_Engine/UtilityMethods.cs

```diff
@@ -0,0 +1,70 @@
+﻿using BH.oM.Base;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Speckle
+{
+    public static class UtilityMethods
+    {
+        public static IEnumerable<IBHoMObject> FilterIBHoMObjects(IEnumerable<object> objects)
+        {
+            List<IBHoMObject> bhomObjects = new List<IBHoMObject>();
+
+            Type iBHoMObjectType = typeof(IBHoMObject);
+
+            MethodInfo miToList = typeof(Enumerable).GetMethod("Cast");
+
+            foreach (var typeGroup in objects.GroupBy(x => x.GetType()))
+            {
+                MethodInfo miListObject = miToList.MakeGenericMethod(new[] { typeGroup.Key });
+
+                var castedObjects = miListObject.Invoke(typeGroup, new object[] { typeGroup });
+
+                if (iBHoMObjectType.IsAssignableFrom(typeGroup.Key))
+                {
+                    bhomObjects.AddRange((IEnumerable<IBHoMObject>)castedObjects);
+                }
+            }
+
+            return bhomObjects;
+        }
+
+        public static void DispatchBHoMObjects(IEnumerable<IObject> objects, out List<IBHoMObject> ibhomObjects, out List<IObject> iObjects)
```

Comment:
As above, `Compute` class?

Reply:
I think it should stay in the Modify as well?

Reply:
Query